src/main/scala/com/cloudant/clouseau/IndexCleanupService.scala

```diff
@@ -51,6 +57,16 @@ class IndexCleanupService(ctx: ServiceContext[ConfigurationArgs]) extends Servic
       cleanup(rootDir, pattern, activeSigs)
   }
 
+  override def handleInfo(msg: Any) = msg match {
+    case ('delete_dir, dir: File) =>
+      if (dir.exists) {
+        val deleted = dir.delete
+        if (!deleted) {
+          self ! ('delete_dir, dir)
```

Comment:
Should there maybe be a small delay here?

Reply:
Oh, I guess not if it blocks general message consumption!
But it seems like we might not want to do this as fast as possible?

Reply:
Yes, I agree that it could be a bit more slower.  But I do not know what to schedule processing of messages for later.

Reply:
Changed the code per @rnewson's suggestion in the meantime.